File: [Manya]/Problem_1.java
public class Problem_1 {
    public int Sum(int[] nums){
        int totalsum = 0;

        for(int num: nums){
            totalsum += num;
        }
        int leftsum = 0;
        for(int i = 0; i < nums.length; i++){
            if(leftsum == totalsum - leftsum - nums[i]){
                return i;
            }
            leftsum += nums[i];
        }
        return -1;
    }

    public static void main(String[] args) {
        int [] nums = {2, 3, -1, 8, 4};
        Problem_1 obj = new Problem_1();

        System.out.println(obj.Sum(nums));
    }
}

File: Ashutosh-Singh/ClassTests/4th-JAN-2024/PROBLEM-1.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    int pivotIndex(vector<int> &nums)
    {
        int n = nums.size();
        int TotalSum = 0;
        for (int i = 0; i < n; i++)
        {
            TotalSum += nums[i];
        }

        int leftSum = 0;
        int rightSum = 0;
        int sum = 0;
        for (int i = 0; i < n; i++)
        {
            sum += nums[i];
            leftSum = sum - nums[i];
            rightSum = TotalSum - sum;
            if (leftSum == rightSum)
            {
                return i;
            }
        }
        return -1;
    }
};

int main() {
    int n;
    cout << "Enter the number of elements in the array: ";
    cin >> n;

    vector<int> nums(n);
    cout << "Enter the elements of the array:\n";
    for (int i = 0; i < n; ++i) {
        cin >> nums[i];
    }

    
    Solution solution;
    int pivot = solution.pivotIndex(nums);

    if (pivot != -1) {
        cout << "Pivot Index: " << pivot << endl;
    } else {
        cout << "No Pivot Index found." << endl;
    }

    return 0;
}
